example/windows/ReactTestAppTests/ManifestTests.cpp

```diff
@@ -0,0 +1,96 @@
+#include "pch.h"
+#include "CppUnitTest.h"
+#include "Manifest.h"
+#include <string>
+
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+using ReactTestApp::Manifest;
+using ReactTestApp::Component;
+
+namespace ReactTestAppTests
+{
+	TEST_CLASS(ManifestTests)
+	{
+	public:
+		TEST_METHOD(ParseManifestWithOneComponent)
+		{
+            std::optional<ReactTestApp::Manifest> manifest = ReactTestApp::GetManifest("manifestTestFiles/simpleManifest.json");
+            if (!manifest.has_value()) {
+                Assert::Fail(L"Couldn't read manifest file");
+            }
+            auto &manifestContents = manifest.value();
+            Assert::IsTrue(manifestContents.name == "Example");
+            Assert::IsTrue(manifestContents.displayName == "Example");
+            Assert::IsTrue(manifestContents.components[0].appKey == "Example");
+            Assert::IsTrue(manifestContents.components[0].displayName == "App");
+            Assert::IsFalse(manifestContents.components[0].initialProperties.has_value());
+		}
+
+        TEST_METHOD(ParseManifestWithMultipleComponents)
+        {
+            std::optional<ReactTestApp::Manifest> manifest =
+                ReactTestApp::GetManifest("manifestTestFiles/withMultipleComponents.json");
+            if (!manifest.has_value()) {
+                Assert::Fail(L"Couldn't read manifest file");
+            }
+            auto &manifestContents = manifest.value();
+            Assert::IsTrue(manifestContents.name == "Example");
+            Assert::IsTrue(manifestContents.displayName == "Example");
+            Assert::IsTrue(manifestContents.components.size() == 2);
```

Comment:
Should we be using the more descriptive asserts like `AreEqual` etc? https://docs.microsoft.com/en-us/visualstudio/test/microsoft-visualstudio-testtools-cppunittestframework-api-reference?view=vs-2019#cppUnitTestAssert_h

Reply:
For some reason `AreEqual` is not so happy about comparing `char *` which is type of expected value and `std::string` which is type of actual value without explicit conversion. We could do something like this if it's still readable
`Assert::AreEqual(manifestContents.name, std::string("Example"));`

Reply:
Can we try using [`std::string_view`](https://en.cppreference.com/w/cpp/string/basic_string_view)? This requires the tests to be compiled with C++17 enabled:
```c++
#include <string_view>
...
using namespace std::literals::string_view_literals;
...
Assert::AreEqual(manifestContents.name, "Example"sv);
```

Reply:
Alternatively, I think the compiler should also be able to infer that you're calling `std::string` constructor with just `{"Example"}`.

Reply:
`std::string_view` doesn't work for me with C++17 enabled. `{"Example"}` works fine.